kustomize/commands/localize/localize.go

```diff
@@ -20,23 +23,26 @@ type arguments struct {
 }
 
 type flags struct {
-	scope string
+	scope    string
+	noVerify bool
 }
 
 // NewCmdLocalize returns a new localize command.
 func NewCmdLocalize(fs filesys.FileSystem) *cobra.Command {
 	var f flags
+	var buildBuffer bytes.Buffer
+	buildCmd := build.NewCmdBuild(fs, &build.Help{}, &buildBuffer)
```

Comment:
How do you feel about refactoring the `build` command to extract the target `RunE` function out of here?
https://github.com/kubernetes-sigs/kustomize/blob/d514df3db0f6074a12bdfece2122ba4154cc31d8/kustomize/commands/build/build.go#L77-L103

It's a lift-and-shift situation and it helps simplifying your invocation, as you wouldn't need to create the full-fledged Cobra command in here.

Reply:
I can do that! It'll be cleaner

Reply:
I tried this out and will still need to create the Cobra command here. I used the Cobra command that is provided in the `RunE` function and passed it down to call a new function I had created:

`localize.go`
```
func callBuildCmd(buffer bytes.Buffer, cmd *cobra.Command, fs filesys.FileSystem, folder string) (buildOutput string, err error) {
	buffer.Reset()
	buildErr := build.RunBuildCmd(cmd, []string{folder}, fs, &buffer)
	if buildErr != nil {
		return "", errors.Wrap(buildErr)
	}

	return buffer.String(), nil
}
```

`build.go`
```
func RunBuildCmd(cmd *cobra.Command, args []string, fSys filesys.FileSystem, writer io.Writer) error {
	if err := Validate(args); err != nil {
		return err
	}
	k := krusty.MakeKustomizer(
		HonorKustomizeFlags(krusty.MakeDefaultOptions(), cmd.Flags()),
	)
	m, err := k.Run(fSys, theArgs.kustomizationPath)
	if err != nil {
		return err
	}
	if theFlags.outputPath != "" && fSys.IsDir(theFlags.outputPath) {
		// Ignore writer; write to o.outputPath directly.
		return MakeWriter(fSys).WriteIndividualFiles(
			theFlags.outputPath, m)
	}
	yml, err := m.AsYaml()
	if err != nil {
		return err
	}
	if theFlags.outputPath != "" {
		// Ignore writer; write to o.outputPath directly.
		return fSys.WriteFile(theFlags.outputPath, yml)
	}
	_, err = writer.Write(yml)
	return err
}
```

but I got the error:
`illegal flag value --reorder ; legal values: [legacy none]`

Reply:
Ah, I missed the flags bit. Sorry about that 😅 I'm gonna take a deeper look to see if there's another way this could be refactored to simplify your invocation.

Reply:
Hi again, @sanaasy, and my deepest apologies for taking this long!

I think a refactor like this could simplify your invocation of `build`:
https://github.com/stormqueen1990/kustomize/blob/ef58ece5c3ad162da8f08a6b9382d85db0909050/kustomize/commands/build/build.go#L114-L135

What do you think about it?
Then you could simply use the `krusty.MakeDefaultOptions()` on your invocation of `RunBuild()` from `localize`.

Reply:
Hey @stormqueen1990! Maybe this cleanup/refactor would work better in a follow up PR, how does that sound?

Reply:
Yes, I think that's fair. I suggested this because I thought it would be more wieldy than having to create the whole Cobra command to invoke the operation.